### PR TITLE
Remove base_path in favor of using script_name from Plug.Conn

### LINKED ISSUE #569: Use conn.script_name for MailboxPreview
Currently one needs to set `base_path` manually for the preview plug to create proper links. But `conn.script_name` does already hold the necessary information.

### PR DESCRIPTION
**Pull request by princemaple**
close #569 
cc @LostKobrakai 

I'm keeping the config because I haven't checked how it works under multiple forward. If you have more information, maybe we can remove it altogether.

### CODE REVIEW

File: lib/plug/mailbox_preview.ex [hunk 1]
```diff
@@ -37,7 +37,7 @@ if Code.ensure_loaded?(Plug) do
     def call(conn, opts) do
       conn =
         conn
-        |> assign(:base_path, opts[:base_path] || "/")
+        |> assign(:base_path, opts[:base_path] || Enum.join(["/" | conn.script_name]))
```
Comment (LostKobrakai) [inline lib/plug/mailbox_preview.ex:40]: Shouldn't this be `Path.join` instead of `Enum.join`?

Comment (princemaple) [reply lib/plug/mailbox_preview.ex:40]: True, that's what went through my head but didn't make it into the code.
Code review wins :)

**Comment by LostKobrakai** [comment]
> I'm keeping the config because I haven't checked how it works under multiple forward.

Afaik `script_name` is meant to accumulate stripped segments exactly so that one can rebuild the proper request path. This is e.g. also done by the phoenix router helpers.

**Comment by princemaple** [comment]
Thanks @LostKobrakai !

**Pull request merged**